lib/OpenQA/Scheduler.pm

```diff
@@ -845,12 +964,66 @@ sub job_duplicate {
         }
     }
 
-    my $clone = $job->duplicate(\%args);
+    # find jobs that must be cloned due to dependencies:
+    # all parents + all running jobs connected with the parents
+    my %to_clone;
+    _job_duplicate_find_parents($job, undef, \%to_clone);
+
+    my $clone;
+
+    # clone the jobs
+    my $jobs = schema->resultset("Jobs")->search(
+        {
+            id => [ keys %to_clone ],
+        }
+    );
+    while (my $j = $jobs->next) {
+        if ($j->id == $job->id) {
+            #the requested job
+            $clone = $to_clone{$j->id}->{clone} = $j->duplicate(\%args);
+        }
+        else {
+            #dependencies
+            $to_clone{$j->id}->{clone} = $j->duplicate();
+        }
+        _job_update_parent($j->id, $to_clone{$j->id}->{clone}->id);
+    }
+
+    # create dependencies for the clones
+    for my $child_id (keys %to_clone) {
+        my $cl_child_id = $to_clone{$child_id}->{clone}->id;
+        for my $parent_id (keys %{$to_clone{$child_id}->{parent}}) {
+            my $cl_parent_id = $parent_id; # scheduled parents were not cloned
+            $cl_parent_id = $to_clone{$parent_id}->{clone}->id if defined $to_clone{$parent_id};
+            schema->resultset("JobDependencies")->create(
+                {
+                    parent_job_id => $cl_parent_id,
+                    child_job_id => $cl_child_id,
+                    dependency => OpenQA::Schema::Result::JobDependencies::PARALLEL,
+                }
+            );
+        }
+    }
+
+    # abort jobs restarted because of dependencies (exclude the original $args{jobid})
+    $jobs = schema->resultset("Jobs")->search(
+        {
+            id => [ keys %to_clone ],
+            state => [OpenQA::Schema::Result::Jobs::EXECUTION_STATES],
+        },
+        {
+            colums => [qw/id worker_id/]
+        }
+    );
+    while (my $j = $jobs->next) {
+        next if $j->id == $job->id;
+        print STDERR "enqueuing abort for ".$j->id." ".$j->worker_id."\n" if $debug;
+        command_enqueue(workerid => $j->worker_id, command => 'abort', job_id => $j->id);
```

Comment:
I added Utils::log_debug in my branch to be able to debug from !controllers. I did some research but was unable to find a meaningful way - because passing controllers or even apps down to models sounds wrong, but still you want to do some debug statement at times